tide/core/ContentWindowController.cpp

```diff
@@ -183,30 +180,54 @@ void ContentWindowController::adjustSize( const SizeState state )
 
     case SIZE_LARGE:
     {
-        const QSizeF wallSize = _displayGroup->getCoordinates().size();
+        const QSizeF wallSize = _displayGroup.getCoordinates().size();
         resize( LARGE_SIZE_SCALE * wallSize, CENTER );
     } break;
 
     case SIZE_FULLSCREEN:
     {
-        _contentWindow->getContent()->resetZoom();
-        QSizeF size = _contentWindow->getContent()->getDimensions();
-        size.scale( _displayGroup->getCoordinates().size(),
+        _contentWindow.getContent()->resetZoom();
+        QSizeF size = _contentWindow.getContent()->getDimensions();
+        size.scale( _displayGroup.getCoordinates().size(),
                     Qt::KeepAspectRatio );
         constrainSize( size );
         const auto fullscreenCoordinates = _getCenteredCoordinates( size );
-        _contentWindow->setFullscreenCoordinates( fullscreenCoordinates );
+        _apply( fullscreenCoordinates );
+    } break;
+
+    case SIZE_FULLSCREEN_MAX:
```

Comment:
why not having only this one? I probably miss something why two fullscreen behaviors are needed